strapi/strapi
# Allow schema builder to set collection name

### What does it do?

Allows the schema builder to use the collectionName passed in instead of always using pluralName

### Why is it needed?

Needed for tests to be able to easily create content types where the plural name differs from the collection name. Also kind of fixes a bug, since it shouldn't be using pluralName if collectionName is available anyway.

When using the `builder` object which accepts a flattened format of a schema (the properties within the 'info' property at the same level as attributes, etc) and creates a real full schema from it, it ignores the collectionName that is set. This means that (for example) when using `builder.addContentType({ singularName: 'sn', 'pluralName:'pl', collectionName:'cn', ...otherRequiredFields})` the resulting schema will have a collection name of 'pl' instead of 'cn'.


### How to test it?

`builder.addContentType({ singularName: 'sn', 'pluralName:'pl', collectionName:'cn', ...otherRequiredFields})` should create a content type with the given collectionName instead of using the pluralName.

It seems like it should just work, and all API and unit tests still pass, but I picked reviewers who are more likely to notice if this may potentially cause a breaking change somewhere that I'm unaware of.

### Related issue(s)/PR(s)
